crates/tangara-lib/src/firmware.rs

```diff
@@ -138,11 +141,21 @@ fn read_images(zip: &mut ZipArchive<File>, firmware: &data::Firmware)
 
         log::debug!("image {} @ {:x?}, {} bytes", image.name, image.addr, data.len());
 
-        images.push(Image {
-            name: image.name.clone(),
-            addr: image.addr,
-            data: data,
-        });
+        let crc32 = zip.by_name(&image.name).unwrap().crc32();
```

Comment:
This `unwrap` can't ever fail because we already call `zip.by_name` in `read_image_data`, but it still gives me the heebie jeebies and I would prefer we rearranged the code so that we don't have to unwrap anything.

What about doing the crc check in `read_image_data`?

Reply:
Oh yes thank you, that reads a lot nicer